### PR TITLE
remove knative from application broker

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- remove all code related to the old knative based eventing

**Related issue(s)**

### CODE REVIEW

File: components/application-broker/go.mod [hunk 1]
```diff
@@ -23,27 +23,25 @@ require (
...
+	golang.org/x/time v0.0.0-20190921001708-c4c64cad1fd0
 	gopkg.in/yaml.v2 v2.3.0
 	istio.io/api v0.0.0-20201113155354-fcf32ac5d223
 	istio.io/client-go v0.0.0-20201113160737-d4c1e2c0a42f
 	istio.io/gogo-genproto v0.0.0-20191029161641-f7d19ec0141d // indirect
 	k8s.io/api v0.18.8
 	k8s.io/apimachinery v0.18.8
 	k8s.io/client-go v0.18.8
-	knative.dev/eventing v0.12.0
 	knative.dev/pkg v0.0.0-20200113182502-b8dc5fbc6d2f
```
Comment (marcobebway) [inline components/application-broker/go.mod:45]: Why do we still need `knative.dev/pkg`?

File: components/application-broker/internal/broker/provision_test.go [hunk 2]
```diff
@@ -190,7 +171,6 @@ func TestProvisionAsync(t *testing.T) {
 func TestProvisionProcessWithKnative(t *testing.T) {
```
Comment (marcobebway) [inline components/application-broker/internal/broker/provision_test.go:171]: Do we still need this test `TestProvisionProcessWithKnative`?

File: components/application-broker/internal/broker/testing/api_object.go [hunk 3]
```diff
@@ -1,148 +1,31 @@
...
-	eventingv1alpha1 "knative.dev/eventing/pkg/apis/eventing/v1alpha1"
-	messagingv1alpha1 "knative.dev/eventing/pkg/apis/messaging/v1alpha1"
-	"knative.dev/pkg/apis"
-	duckv1 "knative.dev/pkg/apis/duck/v1"
 )
 
 const (
-	FakeChannelName      = "fake-chan"
-	FakeSubscriptionName = "fake-sub"
+	FakeChannelName = "fake-chan"
```
Comment (marcobebway) [inline components/application-broker/internal/broker/testing/api_object.go:11]: Do we need this?

File: components/application-broker/internal/broker/testing/api_object.go [hunk 4]
```diff
@@ -1,148 +1,31 @@
...
 const (
-	integrationNamespace                      = "kyma-integration"
-	applicationNameLabelKey                   = "application-name"
-	applicationServiceIDLabelKey              = "application-service-id"
-	brokerNamespaceLabelKey                   = "broker-namespace"
-	knativeEventingInjectionLabelKey          = "knative-eventing-injection"
-	knativeEventingInjectionLabelValueEnabled = "enabled"
-	knSubscriptionNamePrefix                  = "brokersub"
+	integrationNamespace    = "kyma-integration"
+	applicationNameLabelKey = "application-name"
```
Comment (marcobebway) [inline components/application-broker/internal/broker/testing/api_object.go:17]: Do we need these?

File: components/application-broker/go.mod (hunk 1 above)
Comment (k15r) [reply components/application-broker/go.mod:45]: its used in the tests

File: components/application-broker/internal/broker/testing/api_object.go (hunk 3 above)
Comment (k15r) [reply components/application-broker/internal/broker/testing/api_object.go]: no

File: components/application-broker/internal/broker/testing/api_object.go (hunk 4 above)
Comment (k15r) [reply components/application-broker/internal/broker/testing/api_object.go]: nope

File: components/application-broker/internal/broker/provision_test.go (hunk 2 above)
Comment (k15r) [reply components/application-broker/internal/broker/provision_test.go]: its gone

**Review by jasiu001 — APPROVED** [review]

File: components/application-broker/go.mod (hunk 1 above)
Comment (marcobebway) [reply components/application-broker/go.mod:45]: Which tests? What is it testing related to knative?

**Review by marcobebway — APPROVED** [review]

**Review by jasiu001 — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**